app/helpers/provisioning_templates_helper.rb

```diff
@@ -67,6 +67,12 @@ def permitted_actions(template)
     end
   end
 
+  def pxe_with_building_hosts?(template)
+    kinds = ["PXELinux", "PXEGrub"]
+    os_ids = template.operatingsystem_ids
+    !template.template_kind.nil? && kinds.include?(template.template_kind.name) && Host.where(:build => true, :operatingsystem_id => os_ids).count > 0
```

Comment:
Note that using `template_kind` is triggering test failures as it only exists in ProvisioningTemplate, not Ptable.